src/Cli/Microsoft.TemplateEngine.Cli/CliTemplateParameter.cs

```diff
@@ -150,7 +151,24 @@ internal Option GetOption(IReadOnlySet<string> aliases)
                 if (!string.IsNullOrWhiteSpace(DefaultValue)
                     || (Type == ParameterType.String || Type == ParameterType.Choice) && DefaultValue != null)
                 {
-                    option.SetDefaultValue(DefaultValue);
+                    switch (option)
+                    {
+                        case Option<string> stringOption:
+                            stringOption.SetDefaultValue(DefaultValue);
+                            break;
+                        case Option<bool> booleanOption:
+                            booleanOption.SetDefaultValue(bool.Parse(DefaultValue));
+                            break;
+                        case Option<long> integerOption:
+                            integerOption.SetDefaultValue(long.Parse(DefaultValue));
+                            break;
+                        case Option<float> floatOption:
+                            floatOption.SetDefaultValue(float.Parse(DefaultValue));
+                            break;
+                        default:
+                            Debug.Fail($"Unexpected Option type: {option.GetType()}");
+                            break;
+                    }
```

Comment:
imo we should move this code to `GetBaseOption` and do proper validation there. `DefaultValue` may be not a valid type and then this method can throw. It's safer to use `TryParse`.